### ISSUE TITLE
Selenium investigation

### ISSUE THREAD

**Issue by timack** [159356-1702-Extramural/capstone#5]
Selenium, how it works etc.

**Comment by sumnerfit** [comment]
There are 2 parts to Selenium that I can see are relevant to our project.

The first is the in browser IDE.  This allows multiple point and click tests to be "recorded" and then either one or a batch of tests can be run repeatedly to make sure functionality is not compromised.  For example: I wrote a quick test that had 2 pages and a button that jumped between them.  I then clicked the record button (red button, top right) and clicked between the buttons a few times which created the test called SeleniumTest.
![image](https://user-images.githubusercontent.com/22135017/28550127-0ccaa802-7133-11e7-9025-356171f7c204.png)


I then broke the link back to the first page and ran the test again.

![image](https://user-images.githubusercontent.com/22135017/28550178-53161472-7133-11e7-87e9-61e96dfce851.png)

This is run locally and can be used to run a suite of gameplay testing scripts quickly to highlight anything that breaks as we proceed.  There is limited ability for actually traversing a game to completion so we'll need extra code to shortcut to the end for testing how the game finishes.

The second part is a little more involved.  Selenium has the ability to be set up on a remote server so that the game can be tested on multiple browsers across multiple os's to provide comprehensive support testing. This looks to be able to be integrated with Travis-CI as well if that is a path we go down. Rather than point and click, scripts are written to iterate over browsers and OS's so in the timeframe available, this option could be used to identify whether the game breaks on certain browser/os combinations on main game components and not for thorough game evaluation (although it could easily be extended to cover this if it became necessary)
